src/autoscaler.py

```diff
@@ -1,54 +1,99 @@
         node_groups = scale.nodes(default_model=model.cfg)
         if not node_groups:
             logger.info("Missing juju-scale config")
             raise JujuEnvironmentError("Waiting for Juju Configuration")
 
-        provider_args = "--cloud-provider=juju"
-        nodes_args = " ".join([f"--nodes {node}" for node in node_groups])
-        namespace_args = f"--namespace {charm.model.name.strip()}"
+        namespace = f"--namespace {charm.model.name.strip()}"
+        provider = "--cloud-provider=juju"
+        nodes = " ".join([f"--nodes {node}" for node in node_groups])
 
-        self.command = f"{self.binary} {provider_args} {nodes_args} {namespace_args}"
+        extra_args = config["extra_args"].key_values
+        extra = ""
+        if extra_args:
+            extra = " " + " ".join(
+                sorted(f"--{key}='{value}'" for key, value in extra_args.items())
+            )
+
+        self.command = f"{self.binary} {namespace} {provider} {nodes}{extra}"
         return self
 
-    def apply_juju(self, juju_config, charm):
-        self._build_command(juju_config, charm)
+    def apply(self, config, charm):
+        self._build_command(config, charm)
+
         self.secrets = {
-            "user": juju_config["username"],
-            "password": juju_config["password"],
+            "user": config["username"],
+            "password": config["password"],
         }
+
         self.controller_data = {
-            "api-endpoints": juju_config["api_endpoints"].endpoints,
+            "api-endpoints": config["api_endpoints"].endpoints,
             "uuid": str(uuid.uuid4()),  # juju CLI requires a random unique uuid
-            "ca-cert": juju_config["ca_cert"].decoded,
+            "ca-cert": config["ca_cert"].decoded,
         }
+
+        default_model_uuid = config["default_model_uuid"].cfg
+        if default_model_uuid:
+            self.model_data = {
+                "models": {"admin/controller": {}, "admin/default": {"uuid": default_model_uuid}},
+                "current-model": "admin/default",
+            }
+        else:
+            self.model_data = {
+                "models": {"admin/controller": {}},
+                "current-model": "admin/controller",
+            }
+
         missing = {env for env, val in {**self.controller_data, **self.secrets}.items() if not val}
         if missing:
             logger.info("Missing config : %s", ",".join(missing))
             raise JujuEnvironmentError(f"Waiting for Juju Configuration: {','.join(missing)}")
         return self
 
+    def authorize(self, container):
+        container.add_layer(container.name, self.layer, combine=True)
+        container.push(*self.accounts_file, make_dirs=True, **self.root_owned)
+        container.push(*self.controller_file, make_dirs=True, **self.root_owned)
+        container.push(*self.models_file, make_dirs=True, **self.root_owned)
+        self._authorize_juju_cli(container)
```

Comment:
push things into the sidecar container necessary to authorize the juju cli client to communicate with the cluster